docs/platforms/python/integrations/asgi/index__v3.x.mdx

```diff
@@ -0,0 +1,133 @@
+    profile_lifecycle="trace",
+    # ___PRODUCT_OPTION_END___ profiling
+)
+
+app = SentryAsgiMiddleware(app)
+```
+
+The middleware supports both ASGI 2 and ASGI 3 transparently.
+
+## Verify
+
+Trigger an error in your code and see it show up in [sentry.io](https://sentry.io).
+
+```python
+sentry_sdk.init(...) # same as above
+
+def app(scope):
+    async def get_body():
+        return f"The number is: {1/0}" # raises an error!
+
+    async def asgi(receive, send):
+        await send(
+            {
+                "type": "http.response.start",
+                "status": 200,
+                "headers": [[b"content-type", b"text/plain"]],
+            }
+        )
+        await send({"type": "http.response.body", "body": await get_body()})
+
+    return asgi
+
+app = SentryAsgiMiddleware(app)
+```
+
+Run your ASGI app with unicorn:
+
+```bash
+uvicorn main:app --port 8000
+```
+
+Point your browser to [http://localhost:8000](http://localhost:8000) to trigger the error which is then sent to Sentry.
+
+Additionally, a transaction will show up in the "Performance" section on [sentry.io](https://sentry.io).
+
+## Behavior
+
+- Request data is attached to all events: **HTTP method, URL, headers**. Sentry excludes raw bodies and multipart file uploads. Sentry also excludes personally identifiable information (such as user ids, usernames, cookies, authorization headers, IP addresses) unless you set `send_default_pii` to `True`.
+
+- Each request has a separate scope. Changes to the scope within a view, for example setting a tag, will only apply to events sent as part of the request being handled.
+
+- The ASGI middleware does not behave like a regular integration. It is not initialized through an extra parameter to `init` and is not attached to a client. When capturing or supplementing events, it just uses the currently active scopes.
+
+### Default Span Attributes
+
+A set of predefined span attributes will be attached to ASGI transactions by default. These can also be used for sampling since they will also be accessible via the `sampling_context` dictionary in the [`traces_sampler`](/platforms/python/configuration/options/#traces_sampler).
+
+    | Span Attribute                  | Description                  |
+    | ------------------------------- | ---------------------------- |
+    | `network.protocol.name`         | `type` on ASGI scope         |
+    | `url.scheme`                    | `scheme` on ASGI scope       |
+    | `url.path`                      | `path` on ASGI scope         |
+    | `url.query`                     | `query` on ASGI scope        |
+    | `network.protocol.version`      | `http_version` on ASGI scope |
+    | `http.request.method`           | `method` on ASGI scope       |
+    | `server.address`, `server.port` | `server` on ASGI scope       |
+    | `client.address`, `client.port` | `client` on ASGI scope       |
+    | `url.full`                      | full URL                     |
```

Comment:
Maybe worth specifying where we get the full URL from, like we do with the other items here